odoo/addons/base/models/res_company.py

```diff
@@ -194,8 +194,9 @@ def _compute_uninstalled_l10n_module_ids(self):
         self.env.cr.execute("""
             SELECT country.id,
                    ARRAY_AGG(module.id)
-              FROM ir_module_module module,
-                   res_country country
+              FROM ir_module_module module
+              JOIN module_country mc ON mc.module_id = module.id
+              JOIN res_country country ON mc.country_id = country.id
```

Comment:
We don't need to join on `res_country` then.
We can just group by `mc.country_id`, where `mc.country_id IN ANY(%(country_ids)s)`

But then we'd need benchmarks to be able to know what is the best.